src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshalAsMarshallingGeneratorFactory.cs

```diff
@@ -75,8 +75,6 @@ public MarshalAsMarshallingGeneratorFactory(InteropGenerationOptions options, IM
                 case { ManagedType: PointerTypeInfo(_, _, IsFunctionPointer: true), MarshallingAttributeInfo: NoMarshallingInfo or MarshalAsInfo(UnmanagedType.FunctionPtr, _) }:
                     return s_blittable;
 
-                case { ManagedType: SpecialTypeInfo { SpecialType: SpecialType.System_Boolean }, MarshallingAttributeInfo: NoMarshallingInfo }:
```

Comment:
Can we make the diagnostic have a detailed message stating that specifying marshalling info is required for `bool` (similar to what we do for `char`/`string`)?